File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.cpp
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& nums) {
        vector<int> ans(nums.size(), 0);
        int negInd = 1, posInd = 0;
        for (auto num : nums) {
            if (num < 0) {
                ans[negInd] = num;
                negInd += 2;
            } else {
                ans[posInd] = num;
                posInd += 2;
            }
        }
        return ans;
    }
};

File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.java
class Solution {
    public int[] rearrangeArray(int[] nums) {
        int n = nums.length;
        int[] ans = new int[n];
        
        int posInd = 0, negInd = 1;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] >= 0) {
                ans[posInd] = nums[i];
                posInd += 2;
            }
            else {
                ans[negInd] = nums[i];
                negInd += 2;
            }
        }
        return ans;
    }
}